lib/Sema/CSSimplify.cpp

```diff
@@ -6832,6 +6790,61 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
   return SolutionKind::Error;
 }
 
+ConstraintSystem::SolutionKind ConstraintSystem::simplifyTypeMemberConstraint(
+    Type origBaseType, AssociatedTypeDecl *assocType, Type memberType,
+    TypeMatchOptions flags, ConstraintLocatorBuilder locator) {
+  auto formUnsolved = [&] {
+    // If requested, generate a constraint.
+    if (flags.contains(TMF_GenerateConstraints)) {
+      auto *member = Constraint::createTypeMember(
+          *this, origBaseType, memberType, assocType,
+          getConstraintLocator(locator));
+
+      addUnsolvedConstraint(member);
+      return SolutionKind::Solved;
+    }
+
+    return SolutionKind::Unsolved;
+  };
+
+  // FIXME: It's kind of weird in general that we have to look
+  // through lvalue, inout and IUO types here
+  auto baseTy = getFixedTypeRecursive(origBaseType, flags, /*wantRValue=*/true)
```

Comment:
I put this FIXME in a long time ago because I observed DependentMemberTypes where the base was an InOutType of a GenericTypeParamType. This might no longer happen (even without your PR?). Do you want to try removing the call to getWithoutSpecifierType()?

Reply:
Sure! I'll try it out!